src/lib/utilities/popup/types.ts

```diff
@@ -0,0 +1,44 @@
+// Popup Types
+// Note: these are a simple iteration based on the official docs.
+
+// Placement
+// https://floating-ui.com/docs/computePosition#placement
+type Placement =
```

Comment:
through the magic of template literals, you can get this to be a little more concise.

```suggestion
type Direction = 'top' | 'bottom' | 'left' | 'right'
type Placement = Direction | `${Direction}-start` | `${Direction}-end`
```